common/src/main/java/org/wso2/testgrid/common/DashboardSetup.java

```diff
@@ -99,50 +108,68 @@ private String getDataSource(String name) {
+     * This method will create a grafana datasource according to TestPlanID
      */
-    private HttpPost createConnectivity(String restUrl, String apikey) {
-        HttpPost post = new HttpPost(restUrl + ":3000/api/datasources/");
-        post.setHeader("AUTHORIZATION", apikey);
-        post.setHeader("Content-Type", ContentType.APPLICATION_JSON.toString());
-        post.setHeader("Accept", ContentType.APPLICATION_JSON.toString());
-        return post;
-    }
+    public void addGrafanaDataSource() {
 
-    /**
-     * This command will handle the posting http request to REST  api
-     * @param jsonData json string of the data source
-     * @param httpPost post request with authentication
-     */
-    private void executeReq(String jsonData, HttpPost httpPost) {
+        DataOutputStream wr = null;
         try {
-            executeHttpRequest(jsonData, httpPost);
-        } catch (UnsupportedEncodingException e) {
-            logger.error(StringUtil.concatStrings("error while encoding grafana http api url : ", e));
-        } catch (IOException e) {
-            logger.error(StringUtil.concatStrings("ioException occured while sending http request : ", e));
+            String url = "https://" + ConfigurationContext.getProperty(ConfigurationContext.ConfigurationProperties
+                    .GRAFANA_DATASOURCE) + ":3000/api/datasources/";
+
+            HostnameVerifier allHostsValid = (hostname, session) -> true;
+            HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
+
+            URL obj = new URL(url);
+            HttpsURLConnection con = (HttpsURLConnection) obj.openConnection();
+            //add reuqest header
+            con.setRequestMethod("POST");
+            con.setRequestProperty("User-Agent", USER_AGENT);
+            con.setRequestProperty("Content-Type", ContentType.APPLICATION_JSON.toString());
+            con.setRequestProperty("Accept", ContentType.APPLICATION_JSON.toString());
+            con.setRequestProperty("Authorization", apikey);
+            SSLSocketFactory sslSocketFactory = createSslSocketFactory();
+            con.setSSLSocketFactory(sslSocketFactory);
+            String urlParameters = getDataSource(testplanID);
+            con.setDoOutput(true);
+            //logger.info("garafana COnnection: " + con.toString());
+            wr = new DataOutputStream(con.getOutputStream());
+            wr.writeBytes(urlParameters);
+            wr.flush();
+            int responseCode = con.getResponseCode();
+            if (responseCode == 200) {
+                logger.info("grafana Data Source created for testplan " + testplanID);
+            } else {
+                logger.error(StringUtil.concatStrings("failed to create grafana Data testplan ", testplanID,
+                        " Response Code ", responseCode));
+            }
+        } catch (SSLHandshakeException e) {
+            logger.error("Please set a valid truststore key file" + e.toString());
```

Comment:
Is it possible to say exactly that having a invalid truststore key file is the reason for `SSLHandshakeException` ?